File: zukan_workspace/web/ikimono/animal_page.dart
import 'dart:html';
import '../animal.dart';
import 'dart:async';
import 'package:browser_detect/browser_detect.dart';

main() {
  String animal_name = document.baseUri.split("/").last.replaceFirst(".html", "");
  Animal animal = animal_map[animal_name];
  List<String> imageClasses = animal.backgroundImageClasses();
  if (window.screen.width > 700 && imageClasses.length > 1 && browser.isChrome) {
    List<String> backgroundImages = animal.backgroundImages();
    for (String backgroundImage in backgroundImages) {
      new ImageElement(src: backgroundImage); // preload background images
    }
    BodyElement body = document.querySelector("body");
    int currentImageClass = 0;
    Timer timer = new Timer.periodic(new Duration(seconds: 10), (t) {
      body.classes.remove(imageClasses[currentImageClass]);
      currentImageClass = (currentImageClass + 1) % imageClasses.length;
      body.classes.add(imageClasses[currentImageClass]);
    });
  }
}

File: zukan_workspace/web/ikimono/animal_page.js
var isChrome = navigator.userAgent.toLowerCase().indexOf('chrome') > -1;
var body = document.body;
var locations = body.getAttribute("data-locations").split(" ");
if (screen.width > 700 && locations.length > 1 && isChrome) { 
    var location_images = locations.map(function(location) {
        return "/images/seisokuchi/" + location + ".jpg";
    });
    location_images.forEach(function(location_image) {
        var img = new Image();
        img.src = location_image;
    });
    var currentLocation = 0;
    setInterval(function() {
        body.classList.remove(locations[currentLocation]);
        currentLocation = (currentLocation + 1) % locations.length;
        body.classList.add(locations[currentLocation]);
    }, 10000);
}
